### PR TITLE
Don't return window proxy if it has been discarded

### LINKED ISSUE #17479: iframe-append-to-child-document.html fails
```
Append iframe element to its own child document
-----------------------------------------------
Expected PASS, got FAIL
assert_equals: expected null but got object "[object Window]"
@http://web-platform.test:8000/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html:14:3
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1406:20
test@http://web-platform.test:8000/resources/testharness.js:497:9
@http://web-platform.test:8000/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html:8:1
```

### PR DESCRIPTION
**Pull request by cbrewster**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #17479 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17527)

### CODE REVIEW

File: tests/wpt/metadata/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html.ini [hunk 1]
```diff
@@ -2,5 +2,4 @@
   type: testharness
   [Append iframe element to its own child document]
     bug: https://github.com/servo/servo/issues/17479
```
Comment (jdm) [inline tests/wpt/metadata/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html.ini]: We should be able to remove this whole file now.

**Review by jdm — CHANGES_REQUESTED** [review]
Good find!

File: components/script/dom/window.rs [hunk 2]
```diff
@@ -347,6 +347,11 @@ impl Window {
 
     pub fn maybe_window_proxy(&self) -> Option<Root<WindowProxy>> {
```
Comment (asajeffrey) [inline components/script/dom/window.rs]: Bikeshed: rename this to `undiscarded_window_proxy`? There may be some places where the spec will require us to use a window proxy, even if it has been discarded. Might also be worth putting a spec link in to explain what "discarded" means here.

File: tests/wpt/metadata/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html.ini (hunk 1 above)
Comment (asajeffrey) [inline tests/wpt/metadata/html/semantics/embedded-content/the-iframe-element/iframe-append-to-child-document.html.ini]: Yay!

**Review by asajeffrey — COMMENTED** [review]
LGTM, if you address the nitpicking by me and @jdm, you can r=me.

**Comment by cbrewster** [comment]
woops, forgot to add the comment about what discarded means
@bors-servo r=asajeffrey

**Pull request merged**